### PR TITLE
Test fixes

### PR DESCRIPTION
**Pull request by bernardkwok**
- Turn on OGFSX and Arnold generator build and test by default for CI
- Turn on all commented out matrix33 tests
- Fix up Python wrapper for OGSFX
- Make it so if there is a file filter disable implementation count checking.
- Clean up Render test
-- Disable checking of reference OSL implementations which means we don't need to try and string map from OSL to GENOSL.
-- Clean up whitelist
-- Fix up output log names for consistency.

### CODE REVIEW

File: source/MaterialXTest/Render.cpp [hunk 1]
```diff
@@ -1077,12 +1077,18 @@ bool getTestOptions(const std::string& optionFile, ShaderValidTestOptions& optio
             options.renderImages = false;
             options.saveImages = false;
         }
-        // Disable saveing imsages, if no images are to be produced
+        // Disable saving imsages, if no images are to be produced
```
Comment (feldstj) [inline source/MaterialXTest/Render.cpp]: typo: imsages

File: source/MaterialXTest/Render.cpp [hunk 2]
```diff
@@ -1129,39 +1139,41 @@ void printRunLog(const ShaderValidProfileTimes &profileTimes, const ShaderValidT
             {
                 continue;
             }
-            libraryImpls.insert(impl);
+
+            // Only chekc implementations for languages we're interested in and
```
Comment (feldstj) [inline source/MaterialXTest/Render.cpp]: type: chekc

File: source/MaterialXTest/Render.cpp [hunk 3]
```diff
@@ -1119,7 +1125,11 @@ void printRunLog(const ShaderValidProfileTimes &profileTimes, const ShaderValidT
     {
         profilingLog << "---------------------------------------" << std::endl;
 
-        // Get implementation count from libraries
+        const std::string OSL_STRING("osl");
+        const std::string GEN_OSL_STRING(mx::OslShaderGenerator::LANGUAGE);
```
Comment (feldstj) [inline source/MaterialXTest/Render.cpp]: Just wondering why you don't just use mx::OslShaderGenerator::LANGUAGE? Why bother copying the constant?

Comment (bernardkwok) [reply source/MaterialXTest/Render.cpp]: Good point. I'll fix.

**Pull request merged**